src/components/TreeFilter/VirtualizedTreeView.tsx

```diff
@@ -77,21 +78,41 @@ export class VirtualizedTreeView extends React.PureComponent<IVirtualizedTreeVie
             });
             if (scrollToIndex === -1) {
                 // find the top level id of the nested checked item
-                let parent = this.parentLookup[checkedItemIds[0]];   
+                let parent = this.parentLookup[checkedItemIds[0]];
                 while (parent !== undefined) {
                     if (typeof parent.id === 'number') {
                         scrollToIndex = parent.id;
                         break;
                     }
                     parent = this.parentLookup[parent.id];
-                }          
+                }
             }
             this.setState({ scrollToIndex });
         }
     }
 
     public componentWillReceiveProps(nextProps: IVirtualizedTreeViewProps) {
-        if (nextProps.items !== this.props.items) {
+        const newItems = [];
+        if (this.state.searchedItemList !== null) {
+            this.state.searchedItemList.map((data) => {
+                const item = nextProps.items.find(x => x.id === data.id);
+                if (item) {
+                    newItems.push(item);
+                }
+            });
+
+            if (newItems.length !== 0 && newItems !== this.props.items) {

```

Comment:
won't this always be true? newItems = [] on every props receive.

Reply:
Only in case that we have searchedItems it will push those items in newItems. This:
if (newItems.length !== 0 && newItems !== this.props.items) 
first check if we have items in list and if we have then length will be !== 0, if it is just initialized length is 0. 

It's important to keep searched items in newList because on every change of checkbox component receive new props and we don't lose searched items unless we clear search field. It can happen that you searched items -> select all -> deselect all because you maybe want just one of those searched -> and you lose searched item but you still have searched text in field.

Reply:
always as in always in this code block. It will still cause the logic to be triggered each and every time there is something in the searchedItemList.

But i am still puzzled by newItems, searchedItems, filteredItems. What is what and why? 
Why do i need to filter based on newItems instead of props.items? visible items depend only on the text entered.
Why are newItems a parameter in the onItemsSearch callback? are they just put into props.searchedItemList?

Reply:
> It will still cause the logic to be triggered each and every time there is something in the searchedItemList.

Yes, and I think we need it because of the problem that I mentioned. 

> But i am still puzzled by newItems, searchedItems, filteredItems. What is what and why?

I'm going to check if I can somehow refactor it to be less confusing

> Why are newItems a parameter in the onItemsSearch callback?  

When we are calling component VirtualizedTreeView from our project we need to send filterSelecition (which means that we need to have enum (All = 0, None = 1,  Selected = 2)) and I needed it because of that but I think that I can delete it now if I'm checking in ComponentWillRecieveProps. I'll test it.